provider/datagen/src/bin/icu4x-datagen.rs

```diff
@@ -58,6 +64,7 @@ struct Cli {
                     Ignored if '--cldr-root' is present. Requires binary to be built with `networking` Cargo feature (enabled by default).\n\
                     Note that some keys do not support versions before 41.0.0."
     )]
+    #[cfg(feature = "provider")]
```

Comment:
Suggestion: It would be nice if you got an error like `--cldr_tag requires the "provider" and "networking" features` if you try using it without the features. It looks like that might be what you tried doing below with "networking".

Reply:
Yes it would be nice. It would also make this very complex 😭

Reply:
Hmm, I'm not convinced it would be "very" complex. Can you describe why?

Reply:
I think it's a lot more complex because `--cldr-tag` et al have default values (latest), so they're always set.